Content.Client/Chat/ChatManager.cs

```diff
@@ -16,6 +18,20 @@ internal sealed class ChatManager : IChatManager
         private const char OOCAlias = '[';
         private const char MeAlias = '@';
 
+        // Holds any missed messages due to filtering for re-addition to chat later
+<<<<<<< Updated upstream
+        public SortedDictionary<DateTime,MsgChatMessage> filteredHistory = new SortedDictionary<DateTime,MsgChatMessage>();
+=======
+        public List<MsgChatMessage> filteredHistory = new List<MsgChatMessage>();
+>>>>>>> Stashed changes
+
+        // Filter Button states
+        private bool _ALLstate;
+        private bool _OOCstate;
+
+        // List for holding currently filtered channels
+        public List<Enum> filteredChannels = new List<Enum>();
```

Comment:
`ChatChannel` is a `[Flags]` enum for this purpose so you can simply store the filtered channels with a bit field like so:

```cs
private ChatChannel _filteredChannels;
```

For reference: https://stackoverflow.com/questions/8447/what-does-the-flags-enum-attribute-mean-in-c

Also, (this is kind of irrelevant since using a list is quite wasteful anyways), but you shouldn't be storing this as as type `List<Enum>`, you should store it as `List<ChatChannel>`.